test/test_torch.py

```diff
@@ -12340,6 +12340,172 @@ def test_against_np(tensor, bins=100, min=0, max=0):
+        # bins dimension mismatch
+        with self.assertRaisesRegex(RuntimeError, 'custom bin edges must be represented as a one dimensional tensor, '
+                                                  + 'but got a tensor with dimension 2'):
+            torch.histogram(torch.tensor([1], dtype=torch.float, device=device),
+                            bins=torch.tensor([[1, 2], [3, 4]], dtype=torch.float, device=device))
+
+        # empty tensor
+        actual = torch.histogram(torch.tensor([], device=device), range=(0, 3))[0]
+        expected = torch.zeros(10, dtype=torch.long, device=device)
+        self.assertEqual(expected, actual)
+        # empty tensor, bins defined as tensor
+        actual = torch.histogram(torch.tensor([], device=device), bins=torch.tensor([0, 1, 2, 3, 4, 5], device=device))[0]
+        expected = torch.zeros(5, dtype=torch.long, device=device)
+        self.assertEqual(expected, actual)
+        # scalar input and 1 bin -- should return a 1-dimensional tensor, not a scalar.
+        actual = torch.histogram(
+            torch.tensor(0, dtype=torch.float, device=device),
+            bins=1, range=(0, 3))[0]
+        self.assertEqual(
+            torch.tensor([1], dtype=torch.int64, device=device),
+            actual)
+        # tensors with inf; range not provided -- should throw a RuntimeError
+        with self.assertRaisesRegex(RuntimeError, r'range of \[inf, inf\] is not finite'):
+            torch.histogram(torch.tensor([float("inf")], dtype=torch.float, device=device))
+        with self.assertRaisesRegex(RuntimeError, r'range of \[1, inf\] is not finite'):
+            torch.histogram(torch.tensor([1., 2., float("inf")], dtype=torch.float, device=device))
+        # tensors with inf; range provided
+        self.assertEqual(
+            torch.histogram(torch.tensor([float("inf")], dtype=torch.float, device=device),
+                            bins=1, range=(0, 3))[0],
+            torch.tensor([0], dtype=torch.long, device=device))
+        self.assertEqual(
+            torch.histogram(torch.tensor([1., 2., float("inf")], dtype=torch.float, device=device),
+                            bins=4, range=(0, 3))[0],
+            torch.tensor([0, 1, 1, 0], dtype=torch.long, device=device))
+        # tensor with nan -- should throw a RuntimeError
+        with self.assertRaisesRegex(RuntimeError, r'range of \[nan, nan\] is not finite'):
+            torch.histogram(torch.tensor([float("nan")], dtype=torch.float, device=device))
+        # tensors with min > max -- should throw a RuntimeError
+        with self.assertRaisesRegex(RuntimeError, "max must be larger than min"):
+            torch.histogram(torch.tensor([1., 2., 3.], dtype=torch.float, device=device),
+                            bins=4, range=(5, 1))
+
+        # test against numpy.histogram()
+        def test_against_np_uniformbins(tensor, bins=100, min=0, max=0, weights=None, density=False):
```

Comment:
This should be a helper function in a new test called "test_histogram_vs_np_uniform_bins" to improve test granularity and readability.

Reply:
Done.